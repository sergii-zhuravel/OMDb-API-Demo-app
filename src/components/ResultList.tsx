import React from "react";
import styled from 'styled-components';
import ResultItem, { ResultItemProps } from './ResultItem';

const Table = styled.table`
  width: 100%;
`;

export type ResultListProps = {
    result: ResultItemProps[]
}

export default function ResultList({ result = [] }: ResultListProps) {
    return <Table>
        <thead>
            <tr>
                <th>Title</th>
                <th>Year</th>
                <th>Thumbnail</th>
            </tr>
        </thead>
        <tbody>
            {result.map(item => <ResultItem {...item} key={item.imdbID} />)}
        </tbody>
    </Table>
}