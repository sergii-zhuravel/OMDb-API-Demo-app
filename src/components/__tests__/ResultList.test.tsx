import * as React from 'react';
import { shallow } from 'enzyme';
import ResultList from '../ResultList';

test('ResultList component test', () => {
    const mockData = [{ Title: "Title", Year: "2021", Poster: "test/url", imdbID: "1" }];
    const component = shallow(<ResultList result={mockData} />);

    // Snapshot 
    expect(component).toMatchSnapshot();
});