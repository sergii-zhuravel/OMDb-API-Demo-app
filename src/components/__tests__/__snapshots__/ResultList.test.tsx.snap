// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultList component test 1`] = `
<styled.table>
  <thead>
    <tr>
      <th>
        Title
      </th>
      <th>
        Year
      </th>
      <th>
        Thumbnail
      </th>
    </tr>
  </thead>
  <tbody>
    <ResultItem
      Poster="test/url"
      Title="Title"
      Year="2021"
      imdbID="1"
      key="1"
    />
  </tbody>
</styled.table>
`;
